File: 2019/knowit/22/22.js
"use strict";

const fs = require("fs");
const R = require("ramda");

R.compose(
	console.log.bind(console),
	l => l / 5 * 200,
	R.length,
	R.join(""),
	R.map(R.trim),
	R.map(R.join("")),
	R.filter(l => !R.equals(" ", R.last(l))),
	R.transpose,
	R.map(R.split("")),
	R.split("\n")
)(fs.readFileSync("forest.txt", "utf-8"));

File: 2019/knowit/22/22.hs
import Data.List (transpose)

main =
    print
      . (\t -> (t `div` 5) * 200)
      . length
      . filter (/=' ')
      . concat
      . filter ((/=' ') . last)
      . transpose
      . lines
      =<< readFile "forest.txt"
